PureLayout/PureLayout/ALView+PureLayout.m

```diff
@@ -172,13 +179,13 @@ - (NSLayoutConstraint *)autoPinEdgeToSuperviewEdge:(ALEdge)edge withInset:(CGFlo
     ALView *superview = self.superview;
     NSAssert(superview, @"View's superview must not be nil.\nView: %@", self);
     if (edge == ALEdgeBottom || edge == ALEdgeRight || edge == ALEdgeTrailing) {
-        // The bottom, right, and trailing insets (and relations, if an inequality) are inverted to become offsets
-        inset = -inset;
-        if (relation == NSLayoutRelationLessThanOrEqual) {
-            relation = NSLayoutRelationGreaterThanOrEqual;
-        } else if (relation == NSLayoutRelationGreaterThanOrEqual) {
-            relation = NSLayoutRelationLessThanOrEqual;
-        }
+        // This must be formulated a bit differently, with the superview first, in order to keep the constant
+        // positive.
+        self.translatesAutoresizingMaskIntoConstraints = NO;
+        NSLayoutAttribute edgeAttribute = [NSLayoutConstraint al_layoutAttributeForAttribute:(ALAttribute)edge];
+        NSLayoutConstraint *constraint = [NSLayoutConstraint constraintWithItem:superview attribute:edgeAttribute relatedBy:relation toItem:self attribute:edgeAttribute multiplier:1.0 constant:inset];
+        [constraint autoInstall];
+        return constraint;
```

Comment:
A positive inset does not imply that the constant on the underlying constraint will be positive. It's deterministic when the constant will be positive and when it will be negative. And the important point is that this change would be breaking for all existing users because it will cause the constant to flip signs as well as the first and second items on the constraint to swap half the time.

As far as debugging, shouldn't we file a radar so Apple improves this?